rust/src/io/exec/planner.rs

```diff
@@ -186,6 +250,13 @@ impl Planner {
             SQLExpr::BinaryOp { left, op, right } => self.binary_expr(left, op, right),
             SQLExpr::UnaryOp { op, expr } => self.unary_expr(op, expr),
             SQLExpr::Value(value) => self.value(value),
+            // For example, DATE '2020-01-01'
+            SQLExpr::TypedString { data_type, value } => {
```

Comment:
Does it mean we can actually do `INT '1231'` to cast an integer? Is there any restriction on the data_type supported?

Reply:
Could you add a few tests in rust?

Also , should we maintain a doc to describe what expression we support?  Can be a follow up issue tho.

Reply:
Yeah other types work. I think the other one that is good to have is decimal.

I'll add some docs and tests in Rust.